docs/adr/0002-use-json-schema-nimbus-shared.md

```diff
@@ -0,0 +1,63 @@
+# Use JSON Schema generated in nimbus-shared for cross-platform validation of Experiment data
+
+- Status: Accepted
+- Author: Kate Hudson
+- Deciders: Nimbus team, Cirrus team
+- Date: H1 2020
+
+## Context and Problem Statement
+
+Nimbus Experiment configurations are stored in the Experiment Console (Experimenter), sychronized to
+clients via Remote Settings and interpreted by the Nimbus SDK, which as of current writing, includes
+both a JS implementation in Firefox Desktop and a Rust implementation that will eventually replace
+it. Jetstream must also call the Experimenter API in order to read metadata about live experiments.
+
+We need some way to validate data across multiple applications and languages (Python, Rust, JS) and
+a process for publishing changes.
+
+## Decision Outcome
+
+### Format and Language
+
+We decided to use [JSON Schema Draft-07](https://tools.ietf.org/html/draft-handrews-json-schema-01)
+for cross-platform validation of the experiment schema since validators are available in many
+languages, including Python, Rust, and JS.
+
+However, because JSON Schema is difficult to read and write directly, we decided to use Typescript
+as the source for Experiment schemas and generate JSON Schema from them using
+`ts-json-schema-generator`. This library allows us add validations and annotations outside of what
+Typescript provides in its type system (e.g. integers, default and min values, description).
+
+### Publishing changes across the system
+
+The source code for the schema lives in a cental repository
+([mozilla/nimbus-shared](https://github.com/mozilla/nimbus-shared/blob/9a74dde37cbfb73dacd49e8daa66b66cb013619b/types/experiments.ts)).
+Schemas are generated on CircleCI and clients can install the `nimbus-shared` package to access them
```

Comment:
How are we planning on making this available to Rust? Are either the NPM or PyPI distributions accessible to the Rust nimbus-sdk client?

Reply:
I'm asking from my ignorance of Rust's capabilities in regards to pulling in NPM or PyPI packages like this.

Reply:
Filed https://github.com/mozilla/nimbus-shared/issues/136